modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/SCIM2RoleTestCase.java

```diff
@@ -0,0 +1,591 @@
+
+        response = deleteUser(userId4);
+        assertEquals(response.getStatusLine().getStatusCode(), 404, "User " +
+                "has not been deleted successfully");
+        EntityUtils.consume(response.getEntity());
+
+        for (String roleId : roleIdList) { // Delete all roles created.
+            response = sendDeleteUserRequest(roleId);
+            assertEquals(response.getStatusLine().getStatusCode(), 204, "Role " +
+                    "has not been deleted successfully");
+            EntityUtils.consume(response.getEntity());
+            response = sendGetRoleRequest(roleId);
+            assertEquals(response.getStatusLine().getStatusCode(), 404, "Role " +
+                    "has not been deleted successfully");
+            EntityUtils.consume(response.getEntity());
+        }
+    }
+
+    @Test
+    public void testCreateRoleWithUsers() throws Exception {
+
+        JSONObject rootObject = new JSONObject();
+
+        JSONObject user1 = new JSONObject();
+        user1.put(VALUE_PARAM, userId1);
+
+        JSONObject user2 = new JSONObject();
+        user2.put(VALUE_PARAM, userId2);
+
+        JSONArray users = new JSONArray();
+        users.put(user1);
+        users.put(user2);
+
+        rootObject.put(SCIM2BaseTestCase.DISPLAY_NAME_ATTRIBUTE, ROLE_NAME_1);
+        rootObject.put("users", users);
+
+        HttpResponse response = sendCreateRoleRequest(rootObject);
+        assertEquals(response.getStatusLine().getStatusCode(), 201, "Role " +
+                "has not been created successfully");
+
+        JSONObject responseObj = new JSONObject(EntityUtils.toString(response.getEntity()));
+        EntityUtils.consume(response.getEntity());
+
+        String roleNameFromResponse = responseObj.getString(SCIM2BaseTestCase.DISPLAY_NAME_ATTRIBUTE);
+        assertTrue(roleNameFromResponse.contains(ROLE_NAME_1), "Role name is not as expected");
+        roleId1 = responseObj.getString(ID_ATTRIBUTE);
+        assertNotNull(roleId1, "Role id is null");
+        totalRolesCount++;
+
+        HttpResponse getRoleResponse = sendGetRoleRequest(roleId1);
+        assertEquals(getRoleResponse.getStatusLine().getStatusCode(), 200, "Role has not been "
+                + "retrieved successfully");
+        responseObj = new JSONObject(EntityUtils.toString(getRoleResponse.getEntity()));
+        EntityUtils.consume(getRoleResponse.getEntity());
+        JSONArray usersJson = responseObj.getJSONArray("users");
+        assertEquals(usersJson.length(), 2, "Number of users in role is not as expected");
+        assertEquals(userId1, usersJson.getJSONObject(0).getString(VALUE_PARAM), "User id is not as expected");
+        assertEquals(userId2, usersJson.getJSONObject(1).getString(VALUE_PARAM), "User id is not as expected");
+    }
+
+    @Test(dependsOnMethods = "testCreateRoleWithUsers")
+    public void createRolesWithoutUsers() throws Exception {
+
+        for (int i = 2; i < 8; i++) {
```

Comment:
It's not clear about these numbers. Better to move to a constant with a comment.

Reply:
Fixed with d69b33f9e983c84525c179038711cbf05aa9c965